rust/integration-testing/src/bin/flight-test-integration-server.rs

```diff
@@ -0,0 +1,691 @@
+
+        Ok(Response::new(Box::pin(output) as Self::DoGetStream))
+    }
+
+    async fn handshake(
+        &self,
+        _request: Request<Streaming<HandshakeRequest>>,
+    ) -> Result<Response<Self::HandshakeStream>, Status> {
+        Err(Status::unimplemented("Not yet implemented"))
+    }
+
+    async fn list_flights(
+        &self,
+        _request: Request<Criteria>,
+    ) -> Result<Response<Self::ListFlightsStream>, Status> {
+        Err(Status::unimplemented("Not yet implemented"))
+    }
+
+    async fn get_flight_info(
+        &self,
+        request: Request<FlightDescriptor>,
+    ) -> Result<Response<FlightInfo>, Status> {
+        let descriptor = request.into_inner();
+
+        match descriptor.r#type {
+            t if t == DescriptorType::Path as i32 => {
+                let path = &descriptor.path;
+                if path.is_empty() {
+                    return Err(Status::invalid_argument("Invalid path"));
+                }
+
+                let uploaded_chunks = self.uploaded_chunks.lock().await;
+                let flight = uploaded_chunks.get(&path[0]).ok_or_else(|| {
+                    Status::not_found(format!("Could not find flight. {}", path[0]))
+                })?;
+
+                let endpoint = self.endpoint_from_path(&path[0]);
+
+                let total_records: usize =
+                    flight.chunks.iter().map(|chunk| chunk.num_rows()).sum();
+
+                let options = arrow::ipc::writer::IpcWriteOptions::default();
+                let schema = arrow_flight::utils::ipc_message_from_arrow_schema(
+                    &flight.schema,
+                    &options,
+                )
+                .expect(
+                    "Could not generate schema bytes from schema stored by a DoPut; \
+                         this should be impossible",
+                );
+
+                let info = FlightInfo {
+                    schema,
+                    flight_descriptor: Some(descriptor.clone()),
+                    endpoint: vec![endpoint],
+                    total_records: total_records as i64,
+                    total_bytes: -1,
+                };
+
+                Ok(Response::new(info))
+            }
+            other => Err(Status::unimplemented(format!("Request type: {}", other))),
+        }
+    }
+
+    async fn do_put(
+        &self,
+        request: Request<Streaming<FlightData>>,
+    ) -> Result<Response<Self::DoPutStream>, Status> {
+        let mut input_stream = request.into_inner();
+        let flight_data = input_stream
+            .message()
+            .await?
+            .ok_or(Status::invalid_argument("Must send some FlightData"))?;
+
+        let descriptor = flight_data
+            .flight_descriptor
+            .clone()
+            .ok_or(Status::invalid_argument("Must have a descriptor"))?;
+
+        if descriptor.r#type != DescriptorType::Path as i32 || descriptor.path.is_empty()
+        {
```

Comment:
There any rustfmt up in here?

Reply:
yes, rustfmt did this :P also, says the person who didn't run rustfmt on their PR :P